### DISCUSSION TITLE
新手求问：cocos2d的cocos2d::Scheduler::update符号名怎么找？

### DISCUSSION THREAD

**Discussion by android-1995** [bytedance/bhook#58] [Q&A] [answered]
[libcocos2dlua.zip](https://github.com/bytedance/bhook/files/9849941/libcocos2dlua.zip)
```
//尝试用_ZN7cocos2d9Scheduler6updateEf，找不到，status_code=13
bytehook_hook_single("libcocos2dlua.so", nullptr, "_ZN7cocos2d9Scheduler6updateEf", reinterpret_cast<void *>(cos_fun),
                                                      my_hooked_callback, nullptr);
```
有大佬能指点一下吗？

**Comment by caikelun** [comment, marked as answer]
`llvm-readelf -sW ./libcocos2dlua.so | grep Scheduler | grep update`

**Reply by android-1995** [reply]
请问是cmd 运行D:\Android\Sdk\ndk\21.3.6528147\toolchains\llvm\prebuilt\windows-x86_64\arm-linux-androideabi\bin\readelf.exe 执行
`-sW ./libcocos2dlua.so | grep Scheduler | grep update`吗？另外大佬能帮我看看这个方法的符号名具体是多少吗？我想用来校验答案。谢谢大佬了

**Reply by android-1995** [reply]
![image](https://user-images.githubusercontent.com/35674533/197498545-00cf8978-9bac-4ba2-9a68-58fbd6c420ec.png)
windows貌似运行不了grep命令

**Reply by android-1995** [reply]
![image](https://user-images.githubusercontent.com/35674533/197508065-c37edd98-9b0c-44be-a161-e9963c3a79cf.png)
_ZN7cocos2d9Scheduler6updateEf吗？但是hook的时候报13错误码
```
//尝试用_ZN7cocos2d9Scheduler6updateEf，找不到，status_code=13
bytehook_hook_single("libcocos2dlua.so", nullptr, "_ZN7cocos2d9Scheduler6updateEf", reinterpret_cast<void *>(cos_fun),
                                                      my_hooked_callback, nullptr);
```

**Reply by caikelun** [reply]
从你的截图看，_ZN7cocos2d9Scheduler6updateEf 的实现在 libcocos2dlua.so 中。bytehook 是 PLT hook 库，PLT hook是针对函数caller的hook，修改的是函数调用方。如果你需要hook函数本身（调用关系中的callee），可以试试 inlinehook，比如用 shadowhook https://github.com/bytedance/android-inline-hook

**Reply by android-1995** [reply]
好的好的，感谢大佬
